idea_town/frontend/static-src/app/main.js

```diff
@@ -8,33 +12,66 @@ import PageManager from './lib/page-manager';
 import Router from './lib/router';
 
 app.extend({
+
   initialize() {
-    app.experiments = new ExperimentsCollection();
-    app.experiments.fetch();
     app.webChannel = webChannel;
-    app.me = new Me();
-
-    // session won't change without a hard refresh, but addon state could, so:
-    // if addon state changes, dump user back to '/' and let the router handle
-    // redirecting to the correct landing page
-    app.me.on('change:hasAddon', () => { app.router.reload(); });
-
-    // the header is independent of the page container logic, so it lives
-    // outside the page container element
-    // TODO: seems like the PageManager should know when to refresh / hide
-    //       the header
-    app.headerView = new HeaderView({el: document.querySelector('header') });
-    app.headerView.render();
-
-    app.pageManager = new PageManager({
-      pageContainer: document.querySelector('[data-hook=page-container]')
+
+    const apiRoot = this.getDocHeadLink('x-api-root', 'href');
+    let apiRootData;
+    let userData;
+
+    fetch(apiRoot).then((response) => response.json()).then((data) => {
+      apiRootData = data;
+      return fetch(apiRootData.me, { credentials: 'same-origin' });
```

Comment:
It's a little concerning that we have to do two round-trips before we can start rendering the page (since we won't know, until the second fetch returns, whether we have a session or not, hence, whether we need to redirect or render).

I'm fine to leave this for now, provided we file a bug to optimize startup time in the not-too-distant future.